app-template/src/AppInitializer.tsx

```diff
@@ -49,7 +53,45 @@ class AppInitializer extends React.Component<AppInitializerProps, AppInitializer
       </TokenContext.Provider>
     )
   }
+
+  /**
+   * PORTALS-490: Set Synapse callback cookie
+   * Will attempt to set a .synapse.org domain cookie that has enough information to lead the user
+   * back to this portal after visiting www.synapse.org.
+   */
+  updateSynapseCallbackCookie() {
+    let color = 'white'
+    let background = '#4db7ad'
+    let name = ''
+    let icon = ''
+    const footerElement = document.querySelector('#footer')
+    if (footerElement) {
+      color = window.getComputedStyle(footerElement, null).getPropertyValue('color')
+      background = window.getComputedStyle(footerElement, null).getPropertyValue('background-color')
+    }
+    const homeLinkImgElement = document.querySelector('#home-link img')
+    if (homeLinkImgElement) {
+      const imageSrc = homeLinkImgElement.getAttribute('src')
+      if (imageSrc) {
+        icon = imageSrc
+      }
+    }
+    const homeLinkElement = document.querySelector('#home-link')
+    if (homeLinkElement && homeLinkElement.textContent) {
+      name = homeLinkElement.textContent
+    }
+    const cookieValue = {
+      foregroundColor: color,
+      backgroundColor: background,
+      callbackUrl: window.location.href,
+      logoUrl: icon,
+      portalName: name
+    }
+    this.props.cookies.set(
+      'org.sagebionetworks.security.cookies.portal.config',
+      JSON.stringify(cookieValue), { path: '/', domain: '.synapse.org' })
```

Comment:
Should the 'secure' option be provided in the options? https://github.com/reactivestack/cookies/tree/master/packages/react-cookie/#setname-value-options

Reply:
maybe, but not important (not many requests to .synapse.org, and ok to send this silly cookie anywhere)